docs/workflow.md

```diff
@@ -48,6 +48,10 @@ Workflow job summary:
 
 Starting from Tower 3.2, Workflow jobs support simultaneous job runs just like that of ordinary jobs. It is controlled by `allow_simultaneous` field of underlying workflow job template. By default, simultaneous workflow job runs are disabled and users should be prudent in enabling this functionality. Because the performance boost of simultaneous workflow runs will only manifest when a large portion of jobs contained by a workflow allow simultaneous runs. Otherwise it is expected to have some long-running workflow jobs since its spawned jobs can be in pending state for a long time.
 
+Before Tower 3.3, the 'failed' status of workflow job is not defined. Starting from 3.3 we define a finished workflow job to fail, if at least one of the conditions below satisfies:
+* At least one non-leaf node has none of its child nodes spawned to run.
```

Comment:
I need more clarification here, could this mean:

* a node defined a success condition but not a failure condition and then the node failed then it would trigger the success condition?
* what about the scenario where it has a failure condition but no success condition?

Reply:
> At least one non-leaf node has none of its child nodes spawned to run.

This means:
1. If a node runs into success, but there are only failure conditions attached, then the workflow fails.
2. If a node runs into failure, but there are only success conditions attached, then the workflow fails.

Because in both cases, no child node will be triggered.

Reply:
The first of those sounds wrong, unless I'm missing something.

Reply:
Yeah I don't agree with 1. in that list. 2 is a good condition to trigger a failure on but not 1.

Reply:
@matburt @wenottingham Now the rules should be more satisfying:
> A workflow job fails if one of the conditions below satisfies:
> * At least one node runs into states `canceled` or `error`.
> * At least one node runs into states `failed`, but no child node is
spawned to run (no error handler).